_layouts/feedback.html

```diff
@@ -0,0 +1,388 @@
+        var o = [];
+        for(i = 0; i < size; i++){
+            o.push(`hsl(${ i * 360 / size }, 100%, 50%)`)
+        }
+        return o;
+    }
+
+    // Feedback over time
+    function drawFdbkOverMonthGraph(id, datasets) {
+        var fdbkOverMonthGraph = document.getElementById(id);
+        var tutorialsBar = new Chart(fdbkOverMonthGraph, {
+            type: 'line',
+            data: {
+                datasets: datasets
+            },
+            options: {
+                scales: {
+                    yAxes: [{
+                        ticks: {
+                        beginAtZero: true
+                        }
+                    }],
+                    xAxes: [{
+                        type: 'time',
+                        time: {
+                            displayFormats:{month:'YYYY-MM'},
+                            min: '2018-09',
+                            unit: 'month',
+                            distribution: 'linear'
+                        }
+                    }]
+                },
+                legend: {
+                    display: false
+                },
+                title: {
+                    display: true,
+                    text: 'Feedback over time'
+                },
+                plugins: {
+                    colorschemes: {
+                        scheme: 'tableau.Classic20'
+                    }
+                }
+            }
+        });
+    }
+
+    function drawFdbkNoteGraph(id, data, labels) {
+        var tutoBar = document.getElementById(id);
+        var fdbkNoteGraph = new Chart(tutoBar, {
+            type: 'horizontalBar',
+            data: {
+                labels: labels,
+                datasets: [{
+                    backgroundColor: genColors(data.length),
+                    data: data
+                }]
+            },
+            plugins: [ChartDataLabels],
+            options: {
+                legend: {
+                    display: false
+                },
+                title: {
+                    display: true,
+                    text: 'Note distribution'
+                },
+                plugins: {
+                    datalabels: {
+                        anchor: 'end',
+                        align: 'end'
+                    }
+                }
+            }
+        });
+    }
+</script>
+
+<!-- prepare data -->
+{% assign fdbk_by_month = fdbk | group_by: 'month' %}
+{% assign fdbk_over_time = "" | split: ',' %}
+{% assign fdbk_over_time_labels = "" | split: ',' %}
+{% assign fdbk_count = 0 %}
+{% for f in fdbk_by_month %}
+    {% assign new_fdbk = f.items | size %}
+    {% assign fdbk_count = fdbk_count | plus: new_fdbk %}
+    {% assign fdbk_over_time = fdbk_over_time | push: fdbk_count %}
+    {% assign fdbk_over_time_labels = fdbk_over_time_labels | push: f.name %}
+{% endfor %}
+{% assign fdbk_by_note = fdbk | sort:'note' | group_by: 'note' %}
+<script>
+    var fdbk_over_month = {
+        all_topics: [
```

Comment:
would it make sense to have the python script generate these "APIs"? And put the data in the api folder for the javascript to access? It might be a bit easier to maintain than doing it in jekyll (but this looks like a lot of work, totally understand if you just want to keep it.)

Reply:
I was not sure which approach is the best. I did not want to add too much on the GitHub repo but if it makes more sense, let's do that